src/osm2pgsql.cpp

```diff
@@ -63,7 +63,17 @@ int main(int argc, char *argv[])
         auto const outputs =
             output_t::create_outputs(middle->get_query_instance(), options);
 
-        osmdata_t osmdata{middle, outputs};
+        bool const need_dependencies =
+            std::any_of(outputs.cbegin(), outputs.cend(),
+                        [](std::shared_ptr<output_t> const &output) {
+                            return output->need_forward_dependencies();
+                        });
+
+        auto dependency_manager = std::unique_ptr<dependency_manager_t>(
+            need_dependencies ? new full_dependency_manager_t{middle.get()}
+                              : new dependency_manager_t{});
+
+        osmdata_t osmdata{dependency_manager.get(), middle, outputs};
```

Comment:
Any particular reason not to hand ownership over to osmdata? Or rather move the code into the osmdata constructor?

Reply:
Currently the dependency manager doesn't need any configuration. But if it does in the future, so I think it is better created here where the options are available.

Moving into osmdata would be possible, just as with middle and outputs. That's certainly something to consider, but I'd rather do all of them or none of them.

Reply:
I added a commit that moves the dependency manager into osmdata.